### PR TITLE
dbeaver/pro#2151 Most of the items in context menu of file explorer are

### PR DESCRIPTION
**Pull request by serjiokov**
not needed

Fixed unrelated menu items for navigator

### CODE REVIEW

**Comment by serjiokov** [comment]
image

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java [hunk 1]
```diff
@@ -622,4 +622,28 @@ public static void showNodeInNavigator(DBNDatabaseNode dsNode) {
...
+                IContributionItem contribution = (IContributionItem) itemData;
+                String id = contribution.getId();
+                if (id == null) {
+                    continue;
+                }
+                if (id.startsWith("org.eclipse.debug") || // $NON-NLS-0$
+                    id.startsWith("team.main") || // $NON-NLS-0$
+                    id.startsWith("addFromHistoryAction")) { // $NON-NLS-0$
+                    item.dispose();
+                }
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java:639]: This logic breaks incapsulation. dbvfs resources internals has nothing to do with navigator.
Let's use instanceof DBNNodeWithResource. And add new function isRemoteResource there.
All dbfs resources are from DBNPathBase

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/DBNResource.java [hunk 2]
```diff
@@ -626,4 +626,9 @@ public InputStream openInputStream() throws DBException, IOException {
         }
         throw new DBException("Resource '" + getNodeTargetName() + "' doesn't support streaming");
     }
+
+    @Override
+    public boolean isRemoteResource() {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/DBNResource.java:637]: It is always local

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/fs/DBNPathBase.java [hunk 3]
```diff
@@ -473,4 +473,8 @@ public boolean needsInitialization() {
         return children == null;
     }
 
+    @Override
+    public boolean isRemoteResource() {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/fs/DBNPathBase.java:478]: It is always remote

**Review by serge-rider — DISMISSED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/fs/DBNPathBase.java (hunk 3 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/fs/DBNPathBase.java:478]: fixed

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/DBNResource.java (hunk 2 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/DBNResource.java:637]: fixed

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java [hunk 4]
```diff
@@ -622,4 +624,35 @@ public static void showNodeInNavigator(DBNDatabaseNode dsNode) {
...
+                IResource resource = Adapters.adapt(node, IResource.class);
+                if (resource instanceof IFolder &&
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java]:
```java
if ((resource instanceof IFolder) || (node instanceof DBNNodeWithResource nRes && nRes.isRemoteResource()) && 
 id.startsWith("compareWithMenu") || // $NON-NLS-0$
id.startsWith("replaceWithMenu")) { // $NON-NLS-0$
```

I think you can merge the conditions here.

Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java]: Because it looks like code duplicating.

Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java]: I think the better to split out on local-remote resource adaptation branches,
DBNNodeWithResource resource = Adapters.adapt(node, DBNNodeWithResource .class);
if(resouce.isRemoteResource()) {
 // logic for remote
}
else {
 // logic for local
}

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java (hunk 1 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java:639]: Done

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java (hunk 4 above)
Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java]: ok

**Review by LonwoLonwo — APPROVED** [review]

**Review by E1izabeth — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java [hunk 5]
```diff
@@ -622,4 +624,31 @@ public static void showNodeInNavigator(DBNDatabaseNode dsNode) {
...
+                    item.dispose();
+                }
+                DBNNodeWithResource adapt = Adapters.adapt(node, DBNNodeWithResource.class);
+                if (adapt == null) {
+                    return;
+                }
+                if ((adapt.isRemoteResource() ||
+                        adapt.getResource() instanceof IFolder) &&
+                        id.startsWith("compareWithMenu") || // $NON-NLS-0$
+                        id.startsWith("replaceWithMenu")) { // $NON-NLS-0$
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java]: Did you mean to wrap these two checks in parentheses?

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java [hunk 6]
```diff
@@ -622,4 +624,31 @@ public static void showNodeInNavigator(DBNDatabaseNode dsNode) {
...
+        for (MenuItem item : menu.getItems()) {
+            Object itemData = item.getData();
+            if (itemData instanceof IContributionItem contribution) {
+                String id = contribution.getId();
+                if (id == null) {
+                    continue;
+                }
+                if (id.startsWith("org.eclipse.debug") || // $NON-NLS-0$
+                        id.startsWith("team.main") || // $NON-NLS-0$
+                        id.startsWith("addFromHistoryAction")) { // $NON-NLS-0$
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java]: We use 4-space continuation indentation. Please configure in `File | Settings | Editor | Code Style | Java`

Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java]: Fixed

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java (hunk 5 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/NavigatorUtils.java]: done

**Comment by ShadelessFox** [comment]
Please test again against the desktop product and team edition

**Pull request merged**